### PR TITLE
[]15.0][MIG]Hr applicant  Migration 

### PR DESCRIPTION
**Pull request by ilakkia-serpentcs**

### CODE REVIEW

File: hr_applicant/models/hr_emp_trav_lang.py [hunk 1]
```diff
@@ -41,6 +49,10 @@ def _onchange_date(self):
                 "title": _("User Alert !"),
                 "message": _("To Date  must be greater than From Date  !"),
             }
+<<<<<<< HEAD
+=======
+            print("==self.to_date==",self.to_date)
+>>>>>>> 3c4c014ed230e4ae817f9f84e6946f874348091f
```
Comment (JayVora-SerpentCS) [inline hr_applicant/models/hr_emp_trav_lang.py]: !

File: hr_applicant/models/hr_app_med_occ.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            }
+            self.to_date = False
+            return {"warning": warning}
+        if self.from_date and self.to_date and self.from_date > self.to_date:
+            warning = {
+                "title": _("User Alert !"),
+                "message": _("To Date  must be greater than From Date  !"),
+            }
+            self.to_date = False
+            return {"warning": warning}
```
Comment (JayVora-SerpentCS) [inline hr_applicant/models/hr_app_med_occ.py:119]: you could write if common in one line

File: hr_applicant/models/hr_app_rel_edu.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    @api.onchange("birthday")
+    def _onchange_birthday(self):
+        if self.birthday and self.birthday >= fields.Date.today():
+
+            warning = {
+                "title": _("User Alert !"),
+                "message": _("Date of Birth must be less than today!"),
+            }
+            self.birthday = False
+            return {"warning": warning}
```
Comment (JayVora-SerpentCS) [inline hr_applicant/models/hr_app_rel_edu.py:45]: pls see how a return statement can be written

File: hr_applicant/models/hr_app_rel_edu.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    active = fields.Boolean(string="Active", default=True)
+    applicant_id = fields.Many2one("hr.applicant", "Applicant Ref", ondelete="cascade")
+
+    @api.onchange("birthday")
+    def _onchange_birthday(self):
+        if self.birthday and self.birthday >= fields.Date.today():
+
+            warning = {
+                "title": _("User Alert !"),
+                "message": _("Date of Birth must be less than today!"),
```
Comment (JayVora-SerpentCS) [inline hr_applicant/models/hr_app_rel_edu.py:42]: you could make a better text here for user readability

File: hr_applicant/models/hr_app_trav_lang.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
+<<<<<<< HEAD
```
Comment (JayVora-SerpentCS) [inline hr_applicant/models/hr_app_trav_lang.py]: what is this! you cannot see local diff?

File: hr_applicant/models/hr_app_trav_lang.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+from odoo import api, fields, models
+=======
+
+from odoo import _, api, fields, models
+>>>>>>> 3c4c014ed230e4ae817f9f84e6946f874348091f
```
Comment (JayVora-SerpentCS) [inline hr_applicant/models/hr_app_trav_lang.py]: ?

**Comment by JayVora-SerpentCS** [comment]
Ilakkia, this is totally unacceptable as this PR has so many conflicts

**Comment by JayVora-SerpentCS** [comment]
@tushar-serpentcs review plz

File: hr_applicant/models/hr_app_med_occ.py [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    def create(self, vals):
+        if self._context.get("active_model") == "hr.applicant" and self._context.get(
+            "active_id"
+        ):
+            vals.update({"applicant_id": self._context.get("active_id")})
+        return super(ApplicantPreviousOccupation, self).create(vals)
+
+    @api.onchange("from_date", "to_date")
+    def _onchange_date(self):
+
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_app_med_occ.py:104]: Please remove extra space

File: hr_applicant/models/hr_app_med_occ.py [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+            }
+            self.to_date = False
+            return {"warning": warning}
+        if self.from_date and self.to_date and self.from_date > self.to_date:
+            warning = {
+                "title": _("User Alert !"),
+                "message": _("To Date  must be greater than From Date  !"),
+            }
+            self.to_date = False
+            return {"warning": warning}
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_app_med_occ.py:119]: Reference : 
warning['title'] = _('Error!')
if partner:
    warning['message'] = (
        _('%s partner incompatible with relation type.') %
        side.title())
else:
    warning['message'] = (
        _('No %s partner available for relation type.') %
        side)
return warning

File: hr_applicant/models/hr_app_rel_edu.py [hunk 9]
```diff
@@ -0,0 +1,138 @@
+from odoo import api, fields, models
+from odoo.tools.translate import _
+
+
+class ApplicantRelative(models.Model):
+
+    _name = "applicant.relative"
+    _description = "Applicant Relatives"
+    _rec_name = "name"
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_app_rel_edu.py:9]: When you have name field in your columns, you don't need to define that field in _rec_name.

File: hr_applicant/models/hr_app_rel_edu.py [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+            ("Uncle", "Uncle"),
+            ("Wife", "Wife"),
+            ("Other", "Other"),
+        ],
+        string="Relative Type",
+        required=True,
+    )
+    name = fields.Char(string="Name", size=128, required=True)
+    birthday = fields.Date(string="Date of Birth")
+    place_of_birth = fields.Char(string="Place of Birth", size=128)
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_app_rel_edu.py:30]: Size=128 is require ?

File: hr_applicant/models/hr_app_rel_edu.py [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+    def _onchange_illiterate(self):
+        for rec in self:
+            rec.from_date = rec.to_date = rec.country_id = rec.state_id = False
+            rec.education_rank = (
+                rec.school_name
+            ) = rec.grade = rec.field = rec.edu_type = rec.province = ""
+
+    @api.model
+    def create(self, vals):
+
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_app_rel_edu.py:115]: Remove extra space

File: hr_applicant/models/hr_app_rel_edu.py [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+        if self._context.get("active_model") == "hr.applicant" and self._context.get(
+            "active_id"
+        ):
+            vals.update({"applicant_id": self._context.get("active_id")})
+        return super(ApplicantEducation, self).create(vals)
+
+    @api.onchange("from_date", "to_date")
+    def _onchange_date(self):
+
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_app_rel_edu.py:124]: Add description inside of method as per standard

File: hr_applicant/models/hr_app_rel_edu.py [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+            }
+            self.to_date = False
+            return {"warning": warning}
+        if self.from_date and self.to_date and self.from_date > self.to_date:
+            warning = {
+                "title": _("User Alert !"),
+                "message": _("To Date must be greater than From Date !"),
+            }
+            self.to_date = False
+            return {"warning": warning}
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_app_rel_edu.py:138]: Same as above comment

File: hr_applicant/models/hr_app_trav_lang.py [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+    from_date = fields.Date(string="From Date", required=True)
+    to_date = fields.Date(string="To Date", required=True)
+    location = fields.Char(string="Location", size=128, required=True)
+    reason = fields.Char("Reason", required=True)
+    active = fields.Boolean(string="Active", default=True)
+    applicant_id = fields.Many2one("hr.applicant", "Applicant Ref", ondelete="cascade")
+
+    @api.model
+    def create(self, vals):
+
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_app_trav_lang.py:21]: Same as above comment

File: hr_applicant/models/hr_app_trav_lang.py [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+        if self._context.get("active_model") == "hr.applicant" and self._context.get(
+            "active_id"
+        ):
+            vals.update({"applicant_id": self._context.get("active_id")})
+        return super(ApplicantPreviousTravel, self).create(vals)
+
+    @api.onchange("from_date", "to_date")
+    def _onchange_date(self):
+
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_app_trav_lang.py:30]: Same as above comment

File: hr_applicant/models/hr_emp_rel_edu.py [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+    @api.onchange("edu_type")
+    def _onchange_edu_type(self):
+
+        for rec in self:
+            if rec.edu_type == "Local":
+                rec.country_id = False
+            else:
+                rec.province = False
+                rec.state_id = False
+
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_emp_rel_edu.py:107]: Inline

File: hr_applicant/models/hr_emp_rel_edu.py [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+            rec.to_date = False
+            rec.education_rank = ""
+            rec.school_name = ""
+            rec.grade = ""
+            rec.field = ""
+            rec.edu_type = ""
+            rec.country_id = False
+            rec.state_id = False
+            rec.province = ""
+
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_emp_rel_edu.py:121]: Inline

File: hr_applicant/models/hr_recruitment.py [hunk 18]
```diff
@@ -0,0 +1,321 @@
...
+class Applicant(models.Model):
+    _inherit = "hr.applicant"
+    _description = "Applicant"
+
+    @api.depends("medical_ids")
+    def _compute_no_of_medical(self):
+        for rec in self:
+            rec.no_of_medical = len(rec.medical_ids.ids)
+            rec.no_of_medical1 = len(rec.medical_ids.ids)
+
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_recruitment.py:15]: Inline

File: hr_applicant/models/hr_recruitment_employee.py [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+    prev_travel_ids = fields.One2many(
+        "employee.previous.travel", "employee_id", "Previous Travel Ref."
+    )
+    no_of_prev_travel = fields.Integer(
+        "No of Previous Travel", compute="_compute_no_of_prev_travel", readonly=True
+    )
+    lang_ids = fields.One2many("employee.language", "employee_id", "Language Ref.")
+    no_of_lang = fields.Integer(
+        "No of Language", compute="_compute_no_of_lang", readonly=True
+    )
```
Comment (tushar-serpentcs) [inline hr_applicant/models/hr_recruitment_employee.py:69]: New line at the end of the file

File: hr_applicant/wizard/select_training.py [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+
+    @api.onchange("is_triaing_needed")
+    def _onchange_training_courses(self):
+        class_obj = self.env["training.class"]
+        applicant = self.env["hr.applicant"].browse(self._context.get("active_id"))
+        for rec in self:
+            if rec.is_triaing_needed:
+                course = class_obj.search([("job_id", "=", applicant.job_id.id)])
+                if not course:
+
```
Comment (tushar-serpentcs) [inline hr_applicant/wizard/select_training.py:24]: Please remove unnecessary space

**Comment by JayVora-SerpentCS** [comment]
@tushar-serpentcs You may give this to Rohan and let us complete it.

**Comment by tushar-serpentcs** [comment]
Okay sir

**Pull request closed without merging**